src/providers/bedrock/utils.ts

```diff
@@ -45,99 +40,55 @@ export const generateAWSHeaders = async (
-    LLAMA_3_SPECIAL_TOKENS.ROLE_END +
-    '\n';
-  return prompt;
-};
-
-/*
-  This function transforms the messages for the LLama 2 prompt.
-  It combines the system message with the first user message,
-  and then attaches the message pairs.
-  Finally, it adds the last message to the prompt.
-  refer: https://github.com/meta-llama/llama/blob/main/llama/generation.py#L284-L395
-*/
-export const transformMessagesForLLama2Prompt = (messages: Message[]) => {
-  let finalPrompt: string = '';
-  // combine system message with first user message
-  if (messages.length > 0 && messages[0].role === MESSAGE_ROLES.SYSTEM) {
-    messages[0].content =
-      LLAMA_2_SPECIAL_TOKENS.SYSTEM_MESSAGE_START +
-      getMessageContent(messages[0]) +
-      LLAMA_2_SPECIAL_TOKENS.SYSTEM_MESSAGE_END +
-      getMessageContent(messages[1]);
+  if (params['stop']) {
+    inferenceConfig['stopSequences'] = params['stop'];
   }
-  messages = [messages[0], ...messages.slice(2)];
-  // attach message pairs
-  for (let i = 1; i < messages.length; i += 2) {
-    let prompt = getMessageContent(messages[i - 1]);
-    let answer = getMessageContent(messages[i]);
-    finalPrompt += `${LLAMA_2_SPECIAL_TOKENS.BEGINNING_OF_SENTENCE}${LLAMA_2_SPECIAL_TOKENS.CONVERSATION_TURN_START} ${prompt} ${LLAMA_2_SPECIAL_TOKENS.CONVERSATION_TURN_END} ${answer} ${LLAMA_2_SPECIAL_TOKENS.END_OF_SENTENCE}`;
+  if (params['temperature']) {
+    inferenceConfig['temperature'] = params['temperature'];
   }
-  if (messages.length % 2 === 1) {
-    finalPrompt += `${LLAMA_2_SPECIAL_TOKENS.BEGINNING_OF_SENTENCE}${LLAMA_2_SPECIAL_TOKENS.CONVERSATION_TURN_START} ${getMessageContent(messages[messages.length - 1])} ${LLAMA_2_SPECIAL_TOKENS.CONVERSATION_TURN_END}`;
+  if (params['top_p']) {
+    inferenceConfig['topP'] = params['top_p'];
   }
-  return finalPrompt;
+  return inferenceConfig;
 };
 
-/*
-refer: https://docs.mistral.ai/guides/tokenization/
-refer: https://github.com/chujiezheng/chat_templates/blob/main/chat_templates/mistral-instruct.jinja
-*/
-export const transformMessagesForMistralPrompt = (messages: Message[]) => {
-  let finalPrompt: string = `${MISTRAL_CONTROL_TOKENS.BEGINNING_OF_SENTENCE}`;
-  // Mistral does not support system messages. (ref: https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.3/discussions/14)
-  if (messages.length > 0 && messages[0].role === MESSAGE_ROLES.SYSTEM) {
-    messages[0].content =
-      getMessageContent(messages[0]) + '\n' + getMessageContent(messages[1]);
-    messages[0].role = MESSAGE_ROLES.USER;
+export const transformAdditionalModelRequestFields = (
+  params: BedrockChatCompletionsParams
+) => {
+  const additionalModelRequestFields: Record<string, any> =
+    params.additionalModelRequestFields || {};
+  if (params['top_k']) {
+    additionalModelRequestFields['topK'] = params['top_k'];
+  }
+  // Backward compatibility
+  if (params['anthropic_version']) {
+    additionalModelRequestFields['anthropic_version'] =
```

Comment:
As per Bedrock Converse docs, additionalModelRequestFields just needs to have path of the param and not the actual param value:
```
You can specify the paths for additional model parameters in the additionalModelResponseFieldPaths field, as shown in the following example.


[ "/stop_sequence" ]
The API returns the additional fields that you request in the additionalModelResponseFields field.
```

Reply:
Please ignore. Confused `additionalModelRequestFields` with ResponseFields